test/unit/math/opencl/multiply_test.cpp

```diff
@@ -152,9 +151,263 @@ TEST(AgradRevMatrix, multiply_big) {
+                                          stan::math::TriangularViewCL::Lower>(
+      m11, m22);
+
+  stan::math::copy(m3_cl_res, m33);
+
+  EXPECT_MATRIX_NEAR(m3, m3_cl_res, 1e-10);
+}
+
+TEST(MathMatrix, rect_lower_tri_multiply_big) {
+  int size = 512;
+  auto m1 = stan::math::matrix_d::Random(size, size).eval();
+  auto m2 = stan::math::matrix_d::Random(size, size).eval();
+  stan::math::matrix_d m3_cl_res(size, size);
+
+  m2.triangularView<Eigen::StrictlyUpper>().setZero();
+
+  stan::math::matrix_cl m11(m1);
+  stan::math::matrix_cl m22(m2);
+
+  auto m3 = (m1 * m2).eval();
+
+  auto m33 = stan::math::opencl::multiply<stan::math::TriangularViewCL::Entire,
+                                          stan::math::TriangularViewCL::Lower>(
+      m11, m22);
+
+  stan::math::copy(m3_cl_res, m33);
+
+  EXPECT_MATRIX_NEAR(m3, m3_cl_res, 1e-10);
+}
+
+TEST(MathMatrix, rect_lower_tri_multiply_big_rect) {
+  int size = 512;
+  auto m1 = stan::math::matrix_d::Random(size, size).eval();
+  auto m2 = stan::math::matrix_d::Random(size, size * 3).eval();
+  stan::math::matrix_d m3_cl_res(size, size * 3);
+
+  m2.triangularView<Eigen::StrictlyUpper>().setZero();
+
+  stan::math::matrix_cl m11(m1);
+  stan::math::matrix_cl m22(m2);
+
+  auto m3 = (m1 * m2).eval();
+
+  auto m33 = stan::math::opencl::multiply<stan::math::TriangularViewCL::Entire,
+                                          stan::math::TriangularViewCL::Lower>(
+      m11, m22);
+
+  stan::math::copy(m3_cl_res, m33);
+
+  EXPECT_MATRIX_NEAR(m3, m3_cl_res, 1e-10);
+}
+
+TEST(MathMatrix, rect_upper_tri_multiply_small) {
+  auto m1 = stan::math::matrix_d::Random(3, 3).eval();
+  auto m2 = stan::math::matrix_d::Random(3, 3).eval();
+  stan::math::matrix_d m3_cl_res(3, 3);
+
+  m2.triangularView<Eigen::StrictlyLower>().setZero();
+
+  stan::math::matrix_cl m11(m1);
+  stan::math::matrix_cl m22(m2);
+
+  auto m3 = (m1 * m2).eval();
+
+  auto m33 = stan::math::opencl::multiply<stan::math::TriangularViewCL::Entire,
+                                          stan::math::TriangularViewCL::Upper>(
+      m11, m22);
+
+  stan::math::copy(m3_cl_res, m33);
+
+  EXPECT_MATRIX_NEAR(m3, m3_cl_res, 1e-10);
+}
+
+TEST(MathMatrix, rect_upper_tri_multiply_big) {
+  int size = 512;
+  auto m1 = stan::math::matrix_d::Random(size, size).eval();
+  auto m2 = stan::math::matrix_d::Random(size, size).eval();
+  stan::math::matrix_d m3_cl_res(size, size);
+
+  m2.triangularView<Eigen::StrictlyLower>().setZero();
+
+  stan::math::matrix_cl m11(m1);
+  stan::math::matrix_cl m22(m2);
+
+  auto m3 = (m1 * m2).eval();
+
+  auto m33 = stan::math::opencl::multiply<stan::math::TriangularViewCL::Entire,
+                                          stan::math::TriangularViewCL::Upper>(
+      m11, m22);
+
+  stan::math::copy(m3_cl_res, m33);
+
+  EXPECT_MATRIX_NEAR(m3, m3_cl_res, 1e-10);
+}
+
+TEST(MathMatrix, rect_upper_tri_multiply_big_rect) {
+  int size = 512;
```

Comment:
I think we can make a simple tweak to get a little more test coverage: would you mind changing all (except for one) of these test `size`s to some arbitrary numbers that aren't handy powers of 2 or even necessarily even numbers?